### PR TITLE
Disable Tracking

### PR DESCRIPTION
**Pull request by bircni**
In order to make it possible for users to disable tracking, I added an optional bool to set.
If tracking is disabled Aptabase will not be initialized or tracking an event

### CODE REVIEW

**Comment by bircni** [comment]
Reverted the dependency updates as the minSDK jumps from 16 to 19 when updating `androidx.core:core-ktx` or `com.google.android.material` to the newest version

File: aptabase/src/main/java/com/aptabase/Aptabase.kt [hunk 1]
```diff
@@ -43,7 +43,8 @@ class Aptabase private constructor() {
       "SH" to ""
     )
 
-  fun initialize(context: Context, appKey: String, opts: InitOptions? = null) {
+  fun initialize(context: Context, appKey: String, opts: InitOptions? = null, trackingEnabled: Boolean = true) {
```
Comment (cristipufu) [inline aptabase/src/main/java/com/aptabase/Aptabase.kt:46]: Can we add the `trackingEnabled` option to `InitOptions`?
Also, why are the unit tests deleted?

File: aptabase/src/main/java/com/aptabase/Aptabase.kt [hunk 2]
```diff
@@ -55,7 +56,8 @@ class Aptabase private constructor() {
     env = EnvironmentInfo.get(context)
   }
 
-  fun trackEvent(eventName: String, props: Map<String, Any> = emptyMap<String, Any>()) {
+  fun trackEvent(eventName: String, props: Map<String, Any> = emptyMap(), trackingEnabled: Boolean = true) {
```
Comment (cristipufu) [inline aptabase/src/main/java/com/aptabase/Aptabase.kt:59]: i wouldn't add this here, you can skip the method call if tracking is disabled

File: aptabase/src/main/java/com/aptabase/Aptabase.kt (hunk 1 above)
Comment (bircni) [reply aptabase/src/main/java/com/aptabase/Aptabase.kt:46]: Would be an option to put them there...
Unit tests are not necessary and not written so why keeping it?

Comment (cristipufu) [reply aptabase/src/main/java/com/aptabase/Aptabase.kt:46]: Thinking more about this, I don't really find it valuable to add this option to the sdk. You can just wrap the TrackEvent method and handle it yourself as a consumer of the sdk

**Pull request closed without merging**